TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php

```diff
@@ -37,29 +38,54 @@ class DomainCommandController extends \TYPO3\Flow\Cli\CommandController
     protected $siteRepository;
 
     /**
+     * @var ValidatorResolver
+     * @Flow\Inject
+     */
+    protected $validatorResolver;
+
+    /**
      * Add a domain record
      *
      * @param string $siteNodeName The nodeName of the site rootNode, e.g. "neostypo3org"
      * @param string $hostPattern The host pattern to match on, e.g. "neos.typo3.org"
+     * @param string $scheme The scheme for linking (http/https)
+     * @param integer $port The port for linking (0-49151)
      * @return void
      */
-    public function addCommand($siteNodeName, $hostPattern)
+    public function addCommand($siteNodeName, $hostPattern, $scheme = null, $port = null)
     {
         $site = $this->siteRepository->findOneByNodeName($siteNodeName);
         if (!$site instanceof Site) {
-            $this->outputLine('No site found with nodeName "%s".', array($siteNodeName));
+            $this->outputLine('<error>No site found with nodeName "%s".</error>', array($siteNodeName));
             $this->quit(1);
         }
 
         $domains = $this->domainRepository->findByHostPattern($hostPattern);
         if ($domains->count() > 0) {
-            $this->outputLine('The host pattern "%s" is not unique.', array($hostPattern));
+            $this->outputLine('<error>The host pattern "%s" is not unique.</error>', array($hostPattern));
             $this->quit(1);
         }
 
         $domain = new Domain();
+        if ($scheme !== null) {
+            $domain->setScheme($scheme);
+        }
+        if ($port !== null) {
+            $domain->setPort($port);
+        }
         $domain->setSite($site);
         $domain->setHostPattern($hostPattern);
+
+        $domainValidator = $this->validatorResolver->getBaseValidatorConjunction(Domain::class);
+        $result = $domainValidator->validate($domain);
+        if ($result->hasErrors()) {
+            foreach ($result->getFlattenedErrors() as $propertyName => $errors) {
+                $firstError = array_pop($errors);
+                $this->outputLine('<error>Validation failed for "' . $propertyName . '": ' . $firstError . '</error>');
+                $this->quit(1);
+            }
+        }
```

Comment:
Hmm interesting. I get this:

Without:
![image](https://cloud.githubusercontent.com/assets/903567/14455522/4358a91c-00e3-11e6-8090-525514631e2a.png)
With:
![image](https://cloud.githubusercontent.com/assets/903567/14455535/539dd9d2-00e3-11e6-8803-b1b5158f999b.png)